src/opustrainer/modifiers/retokenize.py

```diff
@@ -110,13 +109,16 @@ def __init__(self, probability: float=0.0, src:dict=dict(), trg:dict=dict()):
         self.src = make_retokenizer(src)
         self.trg = make_retokenizer(trg)
 
-    def __call__(self, line:str) -> str:
-        src, trg, alignments = line.split('\t')
-        src_tokens = src.split()
-        trg_tokens = trg.split()
-        pairs = parse_alignments(alignments, src_tokens, trg_tokens)
-        new_src, new_src_tokens, src_mapping = self.src.retokenize(src_tokens)
-        new_trg, new_trg_tokens, trg_mapping = self.trg.retokenize(trg_tokens)
-        remapped_pairs = remap_alignment_pairs(src_mapping, trg_mapping, pairs)
-        return '\t'.join((new_src, new_trg, format_alignments(remapped_pairs)))
-
+    def __call__(self, batch:List[str]) -> Iterable[str]:
+        for line in batch:
```

Comment:
Previously the exception was caught at the Trainer level when it applied the modifier line for line. The Trainer then logged it and skipped that line entirely.

Since we're now submitting batches to modifiers, I've changed these non-critical errors to do the same thing: just skip the line and log about it. Otherwise we'd skip the entire batch (or chunk that was assigned to a specific worker)

I've made sure that logging and raising exceptions still works inside the ModifierWorker processes.

Reply:
So long story short: this modification keeps the behaviour as it is now on the main branch.